app/src/main/java/ch/sdp/vibester/BuzzerScoreUpdater.kt

```diff
@@ -24,15 +27,31 @@ class BuzzerScoreUpdater(ids: ArrayList<Int>, scores: Array<Int>) {
         return buzzerToScoreMap
     }
 
+    fun getWinnerId(): Int {
+        return winnerId
+    }
+
     /**
      * updates the score corresponding to the pressed buzzer
      * checks first if the given id is a buzzer id (and not NO_BUZZER_PRESSED)
      */
-    fun updateScoresArray(id: Int) {
+    fun updateScoresArray(id: Int, point: Int) {
         if (!buzzerToScoreMap.keys.contains(id)) {
             return
         }
-        val updatedScore = buzzerToScoreMap.getOrDefault(id, 0) + 1 // should never get to default,
+        val updatedScore = max(buzzerToScoreMap.getOrDefault(id, 0) + point, 0) // should never get to default
         buzzerToScoreMap.put(id, updatedScore)
+        updateWinner(id, updatedScore)
+    }
+
+    /**
+     * checks if the winner changes after a player scores a point
+     * @param potentialWinnerId: id of the player who scored a point
+     * @param potentialWinnerScore: that player's new score
+     */
+    private fun updateWinner(potentialWinnerId: Int, potentialWinnerScore: Int) {
+        if (winnerId<0 || potentialWinnerScore> buzzerToScoreMap[winnerId]!!) {

```

Comment:
```suggestion
        if (winnerId < 0 || potentialWinnerScore > buzzerToScoreMap[winnerId]!!) {

```
Sorry hahah, this was hard to read, did not want to be mean